spec/src/main/asciidoc/jakarta-ee.adoc

```diff
@@ -1,24 +1,42 @@
+It is the responsibility of the application developer to ensure that each repository is eligible for implementation by exactly one provider, according to the rules specified below.
 
 === Provider support for Entities
 
-A Jakarta Data provider supplies an implementation of repository interfaces in Jakarta Data for one or more types of entities. An entity refers to a class that represents objects in a storage engine, such as SQL or NoSQL databases.
+A Jakarta Data provider supports one or more entity-defining annotation types.
+For example:
 
-The `jakarta.persistence.Entity` annotation from the Jakarta Persistence specification can be used by repository entity classes for Jakarta Data providers that are backed by a Jakarta Persistence provider. Other Jakarta Data providers must not support the `jakarta.persistence.Entity` annotation.
+- The `jakarta.persistence.Entity` annotation from the Jakarta Persistence specification is the entity-defining annotation for Jakarta Data providers backed by a Jakarta Persistence provider. Other Jakarta Data providers must not support the use of `jakarta.persistence.Entity` as an entity-defining annotation.
 
-The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification can be used by repository entity classes for Jakarta Data providers that are backed by NoSQL databases. Other Jakarta Data providers must not support the `jakarta.nosql.Entity` annotation.
+- The `jakarta.nosql.Entity` annotation from the Jakarta NoSQL specification is the entity-defining annotation for Jakarta Data providers backed by NoSQL databases. Other Jakarta Data providers must not support the use of `jakarta.nosql.Entity` as an entity-defining annotation.
 
-Jakarta Data providers that define custom entity annotations must follow the convention that the class name of all supported entity annotation types ends with `Entity`. This enables Jakarta Data providers to identify if a repository entity class contains entity annotations from different Jakarta Data providers so that the corresponding `Repository` can be ignored by Jakarta Data providers that should not provide it.
+A Jakarta Data provider might define a custom entity-defining annotation.
+The name of such a custom entity-defining annotation type must end with `Entity`. This convention allows any Jakarta Data provider to identify when a repository is associated with entity classes declared using an unrecognized entity-defining annotation.
 
-Jakarta Data providers must ignore all `Repository` annotations where annotations for the corresponding entity are available at run time and none of the entity annotations are supported by the Jakarta Data provider. Ignoring these `Repository` annotations allows other Jakarta Data providers to handle them.
+A Jakarta Data provider must ignore every repository associated with entity classes declared using an unrecognized entity-defining annotation, allowing another Jakarta Data provider to supply the implementation.
```

Comment:
There are some subtle, and probably unintentional, changes in meaning here.  The previous language allows for the scenario where an entity class is compiled with multiple entity-defining annotations but only one is available at run time (The TCK actually does this so it can run against either type of provider).  The previous language also allows for a provider that supports a simpler model with unannotated entities.  I believe we can adjust your rewrite as follows to preserve the meaning:
```suggestion
A Jakarta Data provider must ignore every repository associated with entity classes where entity-defining annotations are available at run time but none are recognized by the provider, allowing another Jakarta Data provider to supply the implementation.
```

Reply:
I see. So that scenario wasn't obvious. I would add language explicitly talking about this usage pattern (unless it's already somewhere else in the spec that I missed).

Reply:
I have added language making this more explicit, WDYT, Nathan?